dist/profile/manifests/kubernetes/resources/datadog.pp

```diff
@@ -0,0 +1,62 @@
+# Deploy datadog resources on kubernetes cluster
+class profile::kubernetes::resources::datadog (
+    $apiKey = base64('encode', $::datadog_agent::api_key, 'strict')
+  ){
+  include ::stdlib
+  include profile::kubernetes::params
+  include profile::kubernetes::kubectl
+
+  file { 'datadog_dir':
+    ensure => 'directory',
+    owner  => $profile::kubernetes::params::user,
+    path   => "${profile::kubernetes::params::resources}/datadog",
+  }
+
+  file { 'datadog_secret':
+    ensure  => 'present',
+    content => template("${module_name}/kubernetes/resources/datadog/secret.yaml.erb"),
+    path    => "${profile::kubernetes::params::resources}/datadog/secret.yaml",
+    owner   => $profile::kubernetes::params::user,
+  }
+
+  file { 'datadog_daemonset':
+    ensure  => 'present',
+    content => template("${module_name}/kubernetes/resources/datadog/daemonset.yaml.erb"),
+    path    => "${profile::kubernetes::params::resources}/datadog/daemonset.yaml",
+    owner   => $profile::kubernetes::params::user,
+  }
+
+  $daemonset = loadyaml("${profile::kubernetes::params::resources}/datadog/daemonset.yaml")
+  # Search app label value
+  if ! empty($daemonset){
```

Comment:
IMHO this `if` conditional should have an `else` statement that fails the Puppet run (`fail()` IIRC). If the `$label_app` doesn't get set here, then the "reset pods" will not work properly and fail won't it?